homeassistant/components/husqvarna_automower/sensor.py

```diff
@@ -452,13 +453,34 @@ async def async_setup_entry(
                     for work_area_id in _work_areas
                     if description.exists_fn(_work_areas[work_area_id])
                 )
+                current_work_areas[mower_id] = set(_work_areas)
         entities.extend(
             AutomowerSensorEntity(mower_id, coordinator, description)
             for description in MOWER_SENSOR_TYPES
             if description.exists_fn(coordinator.data[mower_id])
         )
     async_add_entities(entities)
 
+    def _async_work_area_listener() -> None:
+        """Listen for new work areas and add sensor entities if they did not exist."""
+        for mower_id in coordinator.data:
+            if coordinator.data[mower_id].capabilities.work_areas:
+                _work_areas = coordinator.data[mower_id].work_areas
+                if _work_areas is not None:
```

Comment:
```suggestion
            if coordinator.data[mower_id].capabilities.work_areas and (_work_areas := coordinator.data[mower_id].work_areas) is not None:
```
Just a thought, I think this will at least help with the amount of indentation